packages/compass-sidebar/src/components/multiple-connections/saved-connections/saved-connection.tsx

```diff
@@ -274,14 +284,29 @@ export function SavedConnection({
         style={{ visibility: isHovered ? 'visible' : 'hidden' }}
         className={savedConnectionToolbarStyles}
       >
-        <IconButton
-          onClick={() => onConnect(connectionInfo)}
-          data-testid="connect-button"
-          aria-label="Connect"
-          title="Connect"
+        <Tooltip
+          align="top"
+          justify="middle"
+          enabled={!canOpenNewConnection}
+          trigger={({ children, ...props }) => (
+            <IconButton
+              {...props}
+              disabled={!canOpenNewConnection}
+              onClick={() => onConnect(connectionInfo)}
+              data-testid="connect-button"
+              aria-label="Connect"
+              title="Connect"
+            >
+              <Icon glyph="Connect" />
+              {children}
+            </IconButton>
+          )}
         >
-          <Icon glyph="Connect" />
-        </IconButton>
+          {canNotOpenReason === 'maximum-number-exceeded' &&
+            `Only ${maximumNumberOfConnectionsOpen} connection${
+              maximumNumberOfConnectionsOpen > 1 ? 's' : ''
+            } can be open at the same time. First disconnect from another cluster.`}
```

Comment:
Could there be other reason, other than 'maximum-number-exceeded'? Maybe I am not aware of one such but because of that I feel maybe we can exclude this check and just have the string itself :D

Reply:
Yeah there will be other reasons, for example, in the DE this will disabled based on a feature toggle that limits the amount of connections. However, with auto-connect, there will be an option to just disable the thing entirely and the reason would be another (probably 'disabled-by-admin' or something like that).

Reply:
I see, thanks for providing more insights here 👍